packages/core/src/indexing/local-index-api.ts

```diff
@@ -28,16 +29,30 @@ async function _getIndexModelArgs(
     throw new Error(`Cannot index ${modelStreamId.toString()}, it is not a Model StreamID`)
   }
 
+  let opts: IndexModelArgs = {
+    model: modelStreamId,
+  }
+
   if (modelStreamId.type == Model.STREAM_TYPE_ID) {
     const modelState = await repository.load(modelStreamId, {})
     const content = modelState.state.next?.content ?? modelState.state.content
     Model.assertVersionValid(content, 'major')
     if (content.relations) {
-      return { model: modelStreamId, relations: content.relations }
+      opts = {
+        ...opts,
+        relations: content.relations,
+      }
+    }
+    const indices = this.databaseIndexApi.getCompositeIndex(modelStreamId)
```

Comment:
I don't understand the control flow here.  We're in the path that is supposed to be indexing the Model for the first time, but we're checking the database for state about the Model as if it's already indexed?

I'd have expected the list of indices to build would have to be part of the arguments to this function?

Reply:
Okay, I see below you add the index args from the user after (and left a comment about pushing that down into this function).  I still don't understand why we need to check the database here for existing indices though?